2017-11-14 17:33:25,717 INFO  [i.deployment.InternalGameLauncher] [] [] [] : Parsed 362 internal flags at startup
2017-11-14 17:33:25,845 INFO  [i.deployment.InternalGameLauncher] [] [] [] : SpatialOS runtime version: 11-20171114T103507Z-cc9d195
2017-11-14 17:33:25,994 INFO  [i.c.c.f.FlagzFieldRegistryUpdater] [] [] [] : Initialized Fabric flagz registry with 11 set fields.
2017-11-14 17:33:27,099 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.StatusDiscoveryInternalService) on 127.0.0.1:5301.
2017-11-14 17:33:27,100 INFO  [i.c.discovery.ServiceDiscoveryImpl] [] [] [] : Status discovery address: host: "127.0.0.1"
port: 5301

2017-11-14 17:33:27,104 INFO  [improbable.module.ModuleNode] [] [] [] : Starting Fabric node as master (IP: 127.0.0.1).
2017-11-14 17:33:27,105 INFO  [i.c.d.client.CheckpointDiscoveryImpl] [] [] [] : Reached a checkpoint: ALL_NODES_STARTED
2017-11-14 17:33:27,106 INFO  [improbable.module.ModuleNode] [] [] [] : Node master has started, waiting for other nodes to reach the same checkpoint.
2017-11-14 17:33:27,106 INFO  [improbable.module.ModuleNode] [] [] [] : All nodes have been started.
2017-11-14 17:33:27,112 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.runtime.ModuleService) on 127.0.0.1:9000.
2017-11-14 17:33:27,117 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.runtime.HealthCheckService) on 127.0.0.1:9001.
2017-11-14 17:33:27,121 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.PerfInternalService) on 127.0.0.1:9002.
2017-11-14 17:33:27,126 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.ContentionInternalService) on 127.0.0.1:9003.
2017-11-14 17:33:27,139 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.runtime.FeatureService) on 127.0.0.1:9004.
2017-11-14 17:33:27,154 INFO  [i.c.config.flagz.WorkerFlagzUpdater] [] [] [] : UnityWorker worker flag 'max_npc_wizards' updated to value '50'.
2017-11-14 17:33:27,159 INFO  [i.c.config.flagz.WorkerFlagzUpdater] [] [] [] : Initialized worker flagz registry with 1 set fields.
2017-11-14 17:33:27,159 INFO  [improbable.module.ModuleNode] [] [] [] : Using master node configuration loader.
2017-11-14 17:33:27,166 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.NodeConfigurationInternalService) on 127.0.0.1:9005.
2017-11-14 17:33:27,179 INFO  [i.c.c.n.MasterNodeConfigurationLoader] [] [] [] : Using roles [MASTER, GSIM, BRIDGE, WORKER_RUNNER, INSPECTION] for node node_id {
  deployment_id {
    project_id {
      name: "your_project_name_here"
    }
    name: "local"
  }
  name: "master"
}
private_ip: "127.0.0.1"
roles: MASTER
roles: GSIM
roles: BRIDGE
roles: WORKER_RUNNER
roles: INSPECTION

2017-11-14 17:33:27,190 INFO  [improbable.core.config.node.GsimGrid] [] [] [] : Created gsim grid configuration of size (1500, 1500) with 1 columns and 1 rows, using gsims of (30, 30) chunks.
2017-11-14 17:33:27,201 INFO  [i.c.c.n.MasterNodeConfigurationLoader] [] [] [] : Original deployment module configuration (before parsing): {deployment_id {
  project_id {
    name: "your_project_name_here"
  }
  name: "local"
}
name: "master"
=[ProfilingModule, BridgeModuleOracleModule, BridgeOracleModule, PersistenceModule, ReceptionistModule, RouterModule, GSimModule, BridgeModule, EngineStartupModule, InspectionRestModule]}
2017-11-14 17:33:27,211 INFO  [i.c.c.n.MasterNodeConfigurationLoader] [] [] [] : Master successfully loaded configurations for 1 node(s).
2017-11-14 17:33:27,214 INFO  [improbable.module.ModuleNode] [] [] [] : Loaded node configuration: node_id {
  deployment_id {
    project_id {
      name: "your_project_name_here"
    }
    name: "local"
  }
  name: "master"
}
gsim_configurations {
  extent {
    bottom_left {
      x_in_chunks: -15
      z_in_chunks: -15
    }
    top_right {
      x_in_chunks: 15
      z_in_chunks: 15
    }
  }
}
start_router: true
start_bridge_oracle: true
start_bridge_module_oracle: true
start_engine_startup_module: true
start_receptionist_service: true
start_inspection_rest_module: true
start_snapshot_service: true
start_profiling_module: true
bridge_modules: "BridgeModule"
gsim_modules: "GSimModule"

2017-11-14 17:33:27,214 INFO  [improbable.module.ModuleNode] [] [] [] : Using master world extents provider.
2017-11-14 17:33:27,224 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.WorldExtentsInternalService) on 127.0.0.1:9006.
2017-11-14 17:33:27,225 INFO  [i.c.world.MasterWorldExtentsProvider] [] [] [] : Successfully initialized world extents with 1 rectangle(s).
2017-11-14 17:33:27,227 INFO  [improbable.module.ModuleNode] [] [] [] : Loaded world extents: WorldExtents{proto=rectangles {
  bottom_left {
    x: -750.0
    z: -750.0
  }
  top_right {
    x: 750.0
    z: 750.0
  }
  is_fully_simulated: true
  responsible_gsim_id {
  }
}
bounding_rectangle {
  bottom_left {
    x: -750.0
    z: -750.0
  }
  top_right {
    x: 750.0
    z: 750.0
  }
  is_fully_simulated: true
}
}
2017-11-14 17:33:27,327 INFO  [i.e.schema.ParsedFileDescriptorSet] [] [] [] : Parsed file descriptors of 28 proto files.
2017-11-14 17:33:27,353 INFO  [i.entity.schema.SchemaFactory] [] [] [] : Found components: [improbable.fire.Flammable, improbable.player.PlayerControls, improbable.core.UnityWorkerAuthorityCheck, improbable.EntityAcl, improbable.Metadata, improbable.Position, improbable.Persistence, improbable.building.NPCSpawner, improbable.tree.Harvestable, improbable.building.HQInfo, improbable.team.TeamAssignment, improbable.npc.NPCWizard, improbable.core.Inventory, improbable.npc.NPCLumberjack, improbable.building.StockpileDepository, improbable.tree.TreeState, improbable.player.PlayerInfo, improbable.core.ConnectionHeartbeat, improbable.core.ClientAuthorityCheck, improbable.building.BarracksInfo, improbable.core.TransformComponent, improbable.npc.TargetNavigation, improbable.life.Health, improbable.abilities.Spells, improbable.global.PlayerSpawning]
2017-11-14 17:33:27,361 INFO  [i.entity.schema.SchemaFactory] [] [] [] : Found components: [improbable.core.UnityWorkerAuthorityCheck, improbable.npc.NPCLumberjack, improbable.npc.TargetNavigation, improbable.EntityAcl, improbable.Metadata, improbable.Position, improbable.Persistence, improbable.core.ClientAuthorityCheck, improbable.team.TeamAssignment, improbable.building.HQInfo, improbable.building.BarracksInfo, improbable.tree.Harvestable, improbable.player.PlayerInfo, improbable.building.NPCSpawner, improbable.core.Inventory, improbable.tree.TreeState, improbable.player.PlayerControls, improbable.internal.EntityAuthority, improbable.core.TransformComponent, improbable.global.PlayerSpawning, improbable.life.Health, improbable.core.ConnectionHeartbeat, improbable.abilities.Spells, improbable.npc.NPCWizard, improbable.fire.Flammable, improbable.building.StockpileDepository]
2017-11-14 17:33:27,362 INFO  [improbable.module.ModuleNode] [] [] [] : Successfully loaded schema for entities
2017-11-14 17:33:27,363 INFO  [improbable.module.ModuleNode] [] [] [] : Validating snapshot at startup and determining max entity id
2017-11-14 17:33:27,366 INFO  [i.p.s.storage.LegacySnapshotStorage] [] [] [] : Requesting download URL for Snapshot(local@latest) matching []
2017-11-14 17:33:27,380 INFO  [i.p.s.storage.LegacySnapshotStorage] [] [] [] : Downloading snapshot from http://127.0.0.1:21000/snapshots/20171114173253054
2017-11-14 17:33:27,852 INFO  [improbable.module.ModuleNode] [] [] [] : Determined the first entity id for the id assigner as 5568
2017-11-14 17:33:27,855 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.EntityIdInternalService) on 127.0.0.1:9007.
2017-11-14 17:33:27,869 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.WorkerPrestartRegistryInternalService) on 127.0.0.1:9008.
2017-11-14 17:33:27,875 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.RouterLookupInternalService) on 127.0.0.1:9009.
2017-11-14 17:33:27,876 INFO  [i.module.router.ActorSystemFactory$] [] [] [] : Attempting to create ActorSystem akkaFabric with configRecord akkaFabric
2017-11-14 17:33:28,052 INFO  [i.c.d.g.GlobalCheckpointDiscoveryImpl] [] [] [] : Global checkpoint discovery refreshed checkpoints.
2017-11-14 17:33:28,052 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 11 server(s) (previously: 0)
2017-11-14 17:33:28,074 INFO  [akka.event.slf4j.Slf4jLogger] [] [] [] : Slf4jLogger started
2017-11-14 17:33:28,897 INFO  [i.l.params.LoadbalancerParamsFactory] [] [] [] : Parsed 1 worker loadbalancing configuration(s)
2017-11-14 17:33:28,899 INFO  [i.l.params.LoadbalancerParamsFactory] [] [] [] : Successfully loaded loadbalancing configurations: {UnityWorker=improbable.loadbalancing.params.DynamicLoadbalancerParams@6f6363}
2017-11-14 17:33:28,958 INFO  [i.w.a.WorkerAssemblyProviderFactory] [] [] [] : Loaded worker assemblies for these workers types: UnityClient, UnityWorker.
2017-11-14 17:33:28,959 INFO  [improbable.module.ModuleNode] [] [] [] : Loaded worker assemblies: [worker_assembly_id {
  assembly_id {
    project_id {
      name: "your_project_name_here"
    }
    name: "local_assembly"
  }
  name {
    name: "UnityClient"
  }
}
bridge_configuration {
  worker_attribute_set {
    attribute {
      name: "visual"
    }
  }
  entity_interest {
    range_entity_interest {
      radius: 3.0
    }
  }
  streaming_query {
    global_component_streaming_query {
      component_name: "improbable.building.HQInfo"
    }
  }
  component_delivery {
    checkout_all_initially: true
  }
}
, worker_assembly_id {
  assembly_id {
    project_id {
      name: "your_project_name_here"
    }
    name: "local_assembly"
  }
  name {
    name: "UnityWorker"
  }
}
worker_launch_configuration {
  platform: WINDOWS
  worker_zip {
    command: "UnityWorker@Windows.exe"
    argument: "+projectName"
    argument: "${IMPROBABLE_PROJECT_NAME}"
    argument: "+receptionistIp"
    argument: "${IMPROBABLE_RECEPTIONIST_HOST}"
    argument: "+receptionistPort"
    argument: "${IMPROBABLE_RECEPTIONIST_PORT}"
    argument: "+workerType"
    argument: "${IMPROBABLE_WORKER_NAME}"
    argument: "+workerId"
    argument: "${IMPROBABLE_WORKER_ID}"
    argument: "+useInternalIpForBridge"
    argument: "true"
    argument: "+infraServicesUrl"
    argument: "${IMPROBABLE_API_URL}"
    argument: "+assemblyName"
    argument: "${IMPROBABLE_ASSEMBLY_NAME}"
    argument: "+linkProtocol"
    argument: "RakNet"
    argument: "-logfile"
    argument: "${IMPROBABLE_LOG_FILE}"
    argument: "-batchmode"
    argument: "-nographics"
    worker_location {
      filename: "UnityWorker@Windows.zip"
    }
  }
}
worker_launch_configuration {
  platform: MACOS
  worker_zip {
    command: "UnityWorker@Mac.app/Contents/MacOS/UnityWorker@Mac"
    argument: "+projectName"
    argument: "${IMPROBABLE_PROJECT_NAME}"
    argument: "+receptionistIp"
    argument: "${IMPROBABLE_RECEPTIONIST_HOST}"
    argument: "+receptionistPort"
    argument: "${IMPROBABLE_RECEPTIONIST_PORT}"
    argument: "+workerType"
    argument: "${IMPROBABLE_WORKER_NAME}"
    argument: "+workerId"
    argument: "${IMPROBABLE_WORKER_ID}"
    argument: "+useInternalIpForBridge"
    argument: "true"
    argument: "+infraServicesUrl"
    argument: "${IMPROBABLE_API_URL}"
    argument: "+assemblyName"
    argument: "${IMPROBABLE_ASSEMBLY_NAME}"
    argument: "+linkProtocol"
    argument: "RakNet"
    argument: "-logfile"
    argument: "${IMPROBABLE_LOG_FILE}"
    argument: "-batchmode"
    argument: "-nographics"
    worker_location {
      filename: "UnityWorker@Mac.zip"
    }
  }
}
worker_launch_configuration {
  platform: LINUX
  worker_zip {
    command: "UnityWorker@Linux"
    argument: "+projectName"
    argument: "${IMPROBABLE_PROJECT_NAME}"
    argument: "+receptionistIp"
    argument: "${IMPROBABLE_RECEPTIONIST_HOST}"
    argument: "+receptionistPort"
    argument: "${IMPROBABLE_RECEPTIONIST_PORT}"
    argument: "+workerType"
    argument: "${IMPROBABLE_WORKER_NAME}"
    argument: "+workerId"
    argument: "${IMPROBABLE_WORKER_ID}"
    argument: "+useInternalIpForBridge"
    argument: "true"
    argument: "+infraServicesUrl"
    argument: "${IMPROBABLE_API_URL}"
    argument: "+assemblyName"
    argument: "${IMPROBABLE_ASSEMBLY_NAME}"
    argument: "+linkProtocol"
    argument: "RakNet"
    argument: "-logfile"
    argument: "${IMPROBABLE_LOG_FILE}"
    argument: "-batchmode"
    argument: "-nographics"
    worker_location {
      filename: "UnityWorker@Linux.zip"
    }
  }
}
bridge_configuration {
  worker_attribute_set {
    attribute {
      name: "physics"
    }
  }
  entity_interest {
    range_entity_interest {
      radius: 2.0
    }
  }
  component_delivery {
    checkout_all_initially: true
  }
}
]
2017-11-14 17:33:28,962 INFO  [i.w.assembly.WorkerAssemblyProvider] [] [] [] : Worker configuration for 'UnityClient' does not contain a launch configuration for the current platform 'WINDOWS', so Fabric will not start any managed workers of this type
2017-11-14 17:33:28,963 INFO  [i.w.assembly.WorkerAssemblyProvider] [] [] [] : The following worker configurations do not contain valid managed launch configurations for the current platform WINDOWS, and the runtime will be unable to launch managed workers of this type: [UnityClient]. This is expected for worker types you intend to only connect manually.
2017-11-14 17:33:28,964 INFO  [i.w.permissions.PermissionsFactory] [] [] [] : Created permissions manager with permissions: {UnityWorker=Permissions{{/sys/CreateEntity=, /sys/DeleteEntity=, /sys/EntityQuery={"component":["*"]}}}, UnityClient=Permissions{{/sys/EntityQuery={"component":[]}}}}
2017-11-14 17:33:30,060 INFO  [improbable.module.ModuleNode] [] [] [] : Starting router on master
2017-11-14 17:33:30,064 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.RouterInternalService) on 127.0.0.1:9010.
2017-11-14 17:33:30,064 INFO  [i.c.m.r.RouterInternalServiceImpl] [] [] [] : Attempting to register RouterInternalServiceImpl with RouterLookupInternalService.
2017-11-14 17:33:30,070 INFO  [i.c.m.r.l.RouterLookupInternalServiceImpl] [] [] [] : 1/1 router modules started
2017-11-14 17:33:30,071 INFO  [i.c.m.r.RouterInternalServiceImpl] [] [] [] : Registered RouterInternalServiceImpl with RouterLookupInternalService.
2017-11-14 17:33:30,072 INFO  [improbable.module.ModuleNode] [] [] [] : Attempting to get addresses of all RouterModules from master node
2017-11-14 17:33:30,077 INFO  [improbable.module.ModuleNode] [] [] [] : Got all RouterModules from master node
2017-11-14 17:33:30,087 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.RemoteMessageInternalService) on 127.0.0.1:9011.
2017-11-14 17:33:30,095 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.LocalBrokerInternalService) on 127.0.0.1:9012.
2017-11-14 17:33:30,098 INFO  [improbable.module.ModuleNode] [] [] [] : Waiting for all router modules in the deployment to have started
2017-11-14 17:33:30,100 INFO  [improbable.module.ModuleNode] [] [] [] : All router modules in the deployment started
2017-11-14 17:33:30,141 INFO  [org.eclipse.jetty.server.Server] [] [] [] : jetty-8.y.z-SNAPSHOT
2017-11-14 17:33:30,157 INFO  [o.e.jetty.server.AbstractConnector] [] [] [] : Started SelectChannelConnector@0.0.0.0:5006
2017-11-14 17:33:30,158 INFO  [improbable.http.ServletContainer$] [] [] [] : Servlet container started
2017-11-14 17:33:30,159 INFO  [improbable.module.ModuleNode] [] [] [] : Starting BridgeOracle on master
2017-11-14 17:33:30,164 INFO  [improbable.module.ModuleNode] [] [] [] : Waiting for the BridgeOracle to have started
2017-11-14 17:33:30,265 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.BridgeOracleInternalService) on 127.0.0.1:9013.
2017-11-14 17:33:31,057 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 15 server(s) (previously: 11)
2017-11-14 17:33:31,057 INFO  [improbable.module.ModuleNode] [] [] [] : Received BridgeOracle started
2017-11-14 17:33:31,072 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.GlobalBrokerInternalService) on 127.0.0.1:9014.
2017-11-14 17:33:31,073 INFO  [improbable.module.ModuleNode] [] [] [] : Waiting for the BridgeModuleOracle to have started
2017-11-14 17:33:31,077 INFO  [i.c.a.e.ExtensibleActorReporter] [] [] [] : Started BridgeModuleOracleActor(Actor[akka://akkaFabric/user/BridgeModuleOracleModule#41717692]).
2017-11-14 17:33:31,092 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.BridgeServiceOracleInternalService) on 127.0.0.1:9015.
2017-11-14 17:33:32,047 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 17 server(s) (previously: 15)
2017-11-14 17:33:32,047 INFO  [improbable.module.ModuleNode] [] [] [] : Received BridgeModuleOracle started
2017-11-14 17:33:32,051 INFO  [improbable.module.ModuleNode] [] [] [] : Starting gsim GSimModule
2017-11-14 17:33:32,055 INFO  [improbable.module.ModuleNode] [] [] [] : Waiting for 1 entity query service(s) to start.
2017-11-14 17:33:32,076 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.LocalSnapshotInternalService) on 127.0.0.1:9016.
2017-11-14 17:33:32,094 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.fabric.internal.service.ScalingStatsInternalService) on 127.0.0.1:9017.
2017-11-14 17:33:32,097 INFO  [i.p.gsim.GSimPersistenceWriter] [] [] [] : Transitioning GSimPersistenceActor(0) from State [Idle] to [Idle].  Time Elapsed: 0 milliseconds
2017-11-14 17:33:32,105 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.EntityQueryInternalService) on 127.0.0.1:9018.
2017-11-14 17:33:32,281 INFO  [i.bridge.oracle.scaling.WorkerScaler] [] [] [] : Skipping reconciliation because no WorkerRunner service is available. This message is expected at startup but should not persist through the deployment
2017-11-14 17:33:33,054 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 20 server(s) (previously: 17)
2017-11-14 17:33:33,054 INFO  [improbable.module.ModuleNode] [] [] [] : All 1 entity query service(s) have started.
2017-11-14 17:33:33,054 INFO  [improbable.module.ModuleNode] [] [] [] : Creating entity query client.
2017-11-14 17:33:33,101 INFO  [improbable.module.ModuleNode] [] [] [] : Entity query client created with directory: EntityQueryServiceDirectory{allServices=[EntityQueryInternalServiceInfo{address=host: "127.0.0.1"
port: 9018
, extent=Rectangle{minX=-750.0, minZ=-750.0, maxX=750.0, maxZ=750.0}, isLocal=true}], localServices=[EntityQueryInternalServiceInfo{address=host: "127.0.0.1"
port: 9018
, extent=Rectangle{minX=-750.0, minZ=-750.0, maxX=750.0, maxZ=750.0}, isLocal=true}], remoteServices=[]}
2017-11-14 17:33:33,105 INFO  [improbable.module.ModuleNode] [] [] [] : Waiting for 1 bridge service(s) to start.
2017-11-14 17:33:33,117 INFO  [i.c.a.e.ExtensibleActorReporter] [] [] [] : Started BridgeModuleActor(Actor[akka://akkaFabric/user/BridgeModule:3235be05-baa3-4e62-a0b9-5360545f533f#1580052482]).
2017-11-14 17:33:33,137 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.BridgeInternalService) on 127.0.0.1:9019.
2017-11-14 17:33:33,138 INFO  [i.module.bridge.BridgeModuleActor] [] [] [] : Waiting for BridgeModuleOracle service to be available
2017-11-14 17:33:33,145 INFO  [i.module.bridge.BridgeModuleActor] [] [] [] : Registered(BridgeModuleEndpoint(3235be05-baa3-4e62-a0b9-5360545f533f)) on BridgeModuleOracleService
2017-11-14 17:33:34,057 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 21 server(s) (previously: 20)
2017-11-14 17:33:34,058 INFO  [improbable.module.ModuleNode] [] [] [] : All 1 bridge service(s) have started.
2017-11-14 17:33:34,058 INFO  [improbable.module.ModuleNode] [] [] [] : Starting WorkerRunner service (EngineStartupModule)
2017-11-14 17:33:34,068 INFO  [improbable.worker.exec.WorkerStats] [] [] [] : Cannot read /proc/self/stat; worker stats reporting will not work.
2017-11-14 17:33:34,080 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.inner.worker.WorkerRunner) on 127.0.0.1:9020.
2017-11-14 17:33:34,089 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.service.ReceptionistInternalService) on 127.0.0.1:9021.
2017-11-14 17:33:34,098 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.runtime.ReceptionistService) on 127.0.0.1:7777.
2017-11-14 17:33:34,099 INFO  [improbable.module.ModuleNode] [] [] [] : Starting InspectionRestModule
2017-11-14 17:33:34,100 INFO  [improbable.module.ModuleNode] [] [] [] : Starting ProfilingModule
2017-11-14 17:33:34,100 INFO  [i.c.d.client.CheckpointDiscoveryImpl] [] [] [] : Reached a checkpoint: ALL_MODULES_STARTED
2017-11-14 17:33:34,100 INFO  [improbable.module.ModuleNode] [] [] [] : All modules on this node started, waiting for other nodes to reach the same checkpoint.
2017-11-14 17:33:34,103 INFO  [i.c.a.e.ExtensibleActorReporter] [] [] [] : Started ProfilingModuleActor(Actor[akka://akkaFabric/user/ProfilingModule#-954637865]).
2017-11-14 17:33:34,107 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Registered Service(improbable.fabric.internal.debug.HeatmapCellCacheInternalService) on 127.0.0.1:9022.
2017-11-14 17:33:34,125 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.runtime.InspectionService) on 127.0.0.1:9023.
2017-11-14 17:33:34,156 INFO  [i.c.a.e.ExtensibleActorReporter] [] [] [] : Started InspectionRestModuleActor(Actor[akka://akkaFabric/user/InspectionRestModule#-559440919]).
2017-11-14 17:33:34,257 INFO  [i.bridge.oracle.scaling.WorkerScaler] [] [] [] : Skipping reconciliation because no WorkerRunner service is available. This message is expected at startup but should not persist through the deployment
2017-11-14 17:33:34,403 INFO  [spray.can.server.HttpListener] [akka://akkaFabric/user/IO-HTTP/listener-0] [akkaFabric-akka.actor.default-dispatcher-26] [] : Bound to /0.0.0.0:5005
2017-11-14 17:33:35,058 INFO  [improbable.module.ModuleNode] [] [] [] : All modules on all nodes have been started.
2017-11-14 17:33:35,058 INFO  [i.c.d.g.GlobalCheckpointDiscoveryImpl] [] [] [] : Global checkpoint discovery refreshed checkpoints.
2017-11-14 17:33:35,059 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 26 server(s) (previously: 21)
2017-11-14 17:33:35,059 INFO  [improbable.module.ModuleNode] [] [] [] : Loading a snapshot.
2017-11-14 17:33:35,090 INFO  [i.p.g.GSimPersistenceLoadOrchestrator] [] [] [] : Loading snapshot on GSim(0).
2017-11-14 17:33:35,091 INFO  [i.p.s.storage.LegacySnapshotStorage] [] [] [] : Requesting download URL for Snapshot(local@latest) matching []
2017-11-14 17:33:35,098 INFO  [i.p.s.storage.LegacySnapshotStorage] [] [] [] : Downloading snapshot from http://127.0.0.1:21000/snapshots/20171114173253054
2017-11-14 17:33:35,103 INFO  [i.p.gsim.GSimPersistenceLoader] [] [] [] : Loading snapshot
2017-11-14 17:33:35,195 INFO  [i.l.params.LoadbalancerParamsFactory] [] [] [] : LoadbalancerParams for worker type [UnityWorker]: [DynamicLoadbalancerParams{energyLimitingEnabled=true, energyLimit=1.0E9, minWorkerRange=500.0, maxWorkerRange=1000.0, maxWorkerStepSize=200.0, workerExpansionTimeMs=60000, workerGridSize=0, loadbalancerEnabled=true, hexGridMode=OFF, scalerAdvisorType=CONSTANT, maxWorkers=2, initialWorkers=2, numEntitiesPerWorker=0, statusReportPeriodMs=2000, pointsOfInterestEnabled=false, hexGridSolution=Empty, pointsOfInterestSolution=Empty}]
2017-11-14 17:33:37,108 INFO  [i.p.g.GSimPersistenceLoadOrchestrator] [] [] [] : Loaded 5567 entities on GSim(0).
2017-11-14 17:33:37,112 INFO  [i.p.c.PersistenceLoaderCoordinatorActor] [] [] [] : All Snapshots loaded.
2017-11-14 17:33:37,113 INFO  [improbable.module.ModuleNode] [] [] [] : Snapshot has been loaded.
2017-11-14 17:33:37,113 INFO  [improbable.module.ModuleNode] [] [] [] : Starting SnapshotService.
2017-11-14 17:33:37,122 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Registered Service(improbable.platform.runtime.SnapshotService) on 127.0.0.1:9024.
2017-11-14 17:33:37,122 INFO  [i.c.d.client.CheckpointDiscoveryImpl] [] [] [] : Reached a checkpoint: LOAD_SNAPSHOT_FINISHED
2017-11-14 17:33:37,123 INFO  [improbable.module.ModuleNode] [] [] [] : SpatialOS runtime startup complete.
2017-11-14 17:33:38,043 INFO  [i.c.d.g.GlobalCheckpointDiscoveryImpl] [] [] [] : Global checkpoint discovery refreshed checkpoints.
2017-11-14 17:33:38,043 INFO  [i.c.d.g.GlobalServiceDiscoveryImpl] [] [] [] : Global service discovery refreshed 27 server(s) (previously: 26)
2017-11-14 17:33:38,067 INFO  [i.b.o.m.BridgeOracleWorkerManager] [] [] [] : Trying to start managed worker UnityWorker0
2017-11-14 17:33:38,079 INFO  [i.b.o.m.BridgeOracleWorkerManager] [] [] [] : Trying to start managed worker UnityWorker1
2017-11-14 17:33:38,080 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Awaiting internal receptionist service to register with service discovery
2017-11-14 17:33:38,080 INFO  [i.bridge.oracle.scaling.WorkerScaler] [] [] [] : Starting 2 new managed workers for attribute_set {
  attribute: "physics"
}
 (required: 2, existing: 0)
2017-11-14 17:33:38,082 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Successfully found the internal receptionist service in service discovery
2017-11-14 17:33:38,082 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Awaiting internal receptionist service to register with service discovery
2017-11-14 17:33:38,082 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Successfully found the internal receptionist service in service discovery
2017-11-14 17:33:38,085 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Awaiting public receptionist service to register with service discovery
2017-11-14 17:33:38,085 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Awaiting public receptionist service to register with service discovery
2017-11-14 17:33:38,085 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Successfully found the public receptionist service in service discovery
2017-11-14 17:33:38,085 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Successfully found the public receptionist service in service discovery
2017-11-14 17:33:39,206 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Created log file for worker UnityWorker1, type (UnityWorker): [D:\spatialOS - unity -Projects\Astroids\logs\2017-11-14_17-33-05_0\workers\UnityWorker\UnityWorker1-3998399916224566589.log]
2017-11-14 17:33:39,206 INFO  [i.worker.exec.CommandLineExecutor] [] [] [] : Running command: C:\Users\Home\AppData\Local\Temp\zip-downloader8363879239172069732\f7c9bd5fc0f1aa40f9ef84d336172f81\UnityWorker@Windows.exe +projectName your_project_name_here +receptionistIp 127.0.0.1 +receptionistPort 7777 +workerType UnityWorker +workerId UnityWorker1 +useInternalIpForBridge true +infraServicesUrl http://127.0.0.1:21000 +assemblyName local_assembly +linkProtocol RakNet -logfile D:\spatialOS - unity -Projects\Astroids\logs\2017-11-14_17-33-05_0\workers\UnityWorker\UnityWorker1-3998399916224566589.log -batchmode -nographics
2017-11-14 17:33:39,221 INFO  [i.worker.exec.WorkerStartupExecutor] [] [] [] : Created log file for worker UnityWorker0, type (UnityWorker): [D:\spatialOS - unity -Projects\Astroids\logs\2017-11-14_17-33-05_0\workers\UnityWorker\UnityWorker0-9106910389238452358.log]
2017-11-14 17:33:39,222 INFO  [i.worker.exec.CommandLineExecutor] [] [] [] : Running command: C:\Users\Home\AppData\Local\Temp\zip-downloader8363879239172069732\f7c9bd5fc0f1aa40f9ef84d336172f81\UnityWorker@Windows.exe +projectName your_project_name_here +receptionistIp 127.0.0.1 +receptionistPort 7777 +workerType UnityWorker +workerId UnityWorker0 +useInternalIpForBridge true +infraServicesUrl http://127.0.0.1:21000 +assemblyName local_assembly +linkProtocol RakNet -logfile D:\spatialOS - unity -Projects\Astroids\logs\2017-11-14_17-33-05_0\workers\UnityWorker\UnityWorker0-9106910389238452358.log -batchmode -nographics
2017-11-14 17:33:40,276 INFO  [i.b.s.BridgeServicePicker] [] [] [] : Attempting to get positionHint for UnityWorker1 via rpc
2017-11-14 17:33:40,276 INFO  [i.b.s.BridgeServicePicker] [] [] [] : Attempting to get positionHint for UnityWorker0 via rpc
2017-11-14 17:33:40,283 INFO  [i.b.s.BridgeServicePicker] [] [] [] : Selected legacy_guid: "3235be05-baa3-4e62-a0b9-5360545f533f"
address {
  host: "127.0.0.1"
  port: 9019
}
 with position  which is the nearest bridge service to position (510.80837943411007, 0.0, -452.20099814398776)
2017-11-14 17:33:40,283 INFO  [i.b.s.BridgeServicePicker] [] [] [] : Selected legacy_guid: "3235be05-baa3-4e62-a0b9-5360545f533f"
address {
  host: "127.0.0.1"
  port: 9019
}
 with position  which is the nearest bridge service to position (50.99499089206108, 0.0, -32.4492113078353)
2017-11-14 17:33:40,287 INFO  [i.m.b.o.m.BridgeModuleOracleActor] [] [] [] : BridgeServicePicker returned module Optional[legacy_guid: "3235be05-baa3-4e62-a0b9-5360545f533f"
address {
  host: "127.0.0.1"
  port: 9019
}
] with load Optional[BridgeModuleLoadBalanceInfo((0.0, 0.0, 0.0),0)].get().toString() for Engine(UnityWorker0)
2017-11-14 17:33:40,287 INFO  [i.m.b.o.m.BridgeModuleOracleActor] [] [] [] : BridgeServicePicker returned module Optional[legacy_guid: "3235be05-baa3-4e62-a0b9-5360545f533f"
address {
  host: "127.0.0.1"
  port: 9019
}
] with load Optional[BridgeModuleLoadBalanceInfo((0.0, 0.0, 0.0),0)].get().toString() for Engine(UnityWorker1)
2017-11-14 17:33:40,498 INFO  [i.b.i.BridgeInternalServiceImpl] [] [] [] : Bridge worker_id: "UnityWorker0"
 registered with bridge internal service
2017-11-14 17:33:40,498 INFO  [i.b.i.BridgeInternalServiceImpl] [] [] [] : Bridge worker_id: "UnityWorker1"
 registered with bridge internal service
2017-11-14 17:33:40,505 INFO  [i.b.o.s.BridgeOracleInternalServiceImpl] [] [] [] : Got bridge started rpc from bridge: worker_id: "UnityWorker1"

2017-11-14 17:33:40,505 INFO  [i.b.o.s.BridgeOracleInternalServiceImpl] [] [] [] : The worker UnityWorker1 registered with SpatialOS successfully.
2017-11-14 17:33:40,505 INFO  [i.module.bridge.oracle.BridgeOracle] [] [] [] : Sending bridge started to self for bridge endpoint BridgeOracleEndpoint
2017-11-14 17:33:40,506 INFO  [i.module.bridge.oracle.BridgeOracle] [] [] [] : Bridge(UnityWorker1) started.
2017-11-14 17:33:40,510 INFO  [i.b.o.m.BridgeOracleWorkerManager] [] [] [] : Worker UnityWorker1 registered with the BridgeOracleWorker manager
2017-11-14 17:33:40,517 INFO  [i.b.o.s.BridgeOracleInternalServiceImpl] [] [] [] : Got bridge started rpc from bridge: worker_id: "UnityWorker0"

2017-11-14 17:33:40,517 INFO  [i.b.o.s.BridgeOracleInternalServiceImpl] [] [] [] : The worker UnityWorker0 registered with SpatialOS successfully.
2017-11-14 17:33:40,517 INFO  [i.module.bridge.oracle.BridgeOracle] [] [] [] : Sending bridge started to self for bridge endpoint BridgeOracleEndpoint
2017-11-14 17:33:40,518 INFO  [i.module.bridge.oracle.BridgeOracle] [] [] [] : Bridge(UnityWorker0) started.
2017-11-14 17:33:40,518 INFO  [i.b.o.m.BridgeOracleWorkerManager] [] [] [] : Worker UnityWorker0 registered with the BridgeOracleWorker manager
2017-11-14 17:33:41,304 INFO  [i.b.n.raknet.GuardedRakPeerInterface] [] [] [] : New RakNet connection established with 127.0.0.1
2017-11-14 17:33:41,380 INFO  [i.b.n.raknet.GuardedRakPeerInterface] [] [] [] : New RakNet connection established with 127.0.0.1
2017-11-14 17:33:52,265 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker1] HQ failed to spawn barracks due to timeout.
 in  -[WorkerLogger:Unity]
2017-11-14 17:33:53,481 INFO  [i.b.client.BridgeServiceTaskRetrier] [] [] [] : An internal operation forwarding EntityCommandRequest for "entity_id: 3690
component_id: 1601
" failed on the SpatialOS side of the connection to UnityWorker1 after 0 retries, is the deployment under heavy load?
io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:442) [fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) [fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) [fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
2017-11-14 17:33:53,486 WARN  [i.b.commands.BridgeCommandHandler] [] [] [] : Failed to execute command entity_command_request {
  endpoint {
    entity_id: 3690
    component_id: 1601
  }
  payload: "\212d\004\n\002\b*"
}
 on UnityWorker1
improbable.bridge.commands.exception.CommandException: DEADLINE_EXCEEDED
	at improbable.bridge.commands.exception.CommandException.ofStatusRuntimeException(CommandException.java:76) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.commands.exception.CommandException.of(CommandException.java:27) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.client.BridgeInternalServiceClient.lambda$invokeCommand$2(BridgeInternalServiceClient.java:93) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:239) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:229) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:130) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.set(AbstractFuture.java:664) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractCatchingFuture$CatchingFuture.setResult(AbstractCatchingFuture.java:210) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractCatchingFuture.run(AbstractCatchingFuture.java:138) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:112) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.SettableFuture.setException(SettableFuture.java:54) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.client.BridgeServiceTaskRetrier.onFailure(BridgeServiceTaskRetrier.java:102) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1228) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:466) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:442) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) ~[fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) ~[fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_131]
Caused by: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	... 13 common frames omitted
2017-11-14 17:33:53,894 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker1] NPC failed to receive Harvest response
 in  -[WorkerLogger:Unity]
2017-11-14 17:33:53,951 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker1] Received incoming command response for unknown request ID 0. This is usually because a response was received after the command timed out locally.
 in  -[WorkerLogger:Unity]
2017-11-14 17:33:54,514 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker0] NPC failed to receive Harvest response
 in  -[WorkerLogger:Unity]
2017-11-14 17:33:54,718 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker1] Received incoming command response for unknown request ID 1. This is usually because a response was received after the command timed out locally.
 in  -[WorkerLogger:Unity]
2017-11-14 17:33:54,830 INFO  [i.b.client.BridgeServiceTaskRetrier] [] [] [] : An internal operation forwarding EntityCommandRequest for "entity_id: 3690
component_id: 1601
" failed on the SpatialOS side of the connection to UnityWorker0 after 0 retries, is the deployment under heavy load?
io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:442) [fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) [fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) [fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
2017-11-14 17:33:54,833 WARN  [i.b.commands.BridgeCommandHandler] [] [] [] : Failed to execute command entity_command_request {
  endpoint {
    entity_id: 3690
    component_id: 1601
  }
  payload: "\212d\004\n\002\b*"
}
 on UnityWorker0
improbable.bridge.commands.exception.CommandException: DEADLINE_EXCEEDED
	at improbable.bridge.commands.exception.CommandException.ofStatusRuntimeException(CommandException.java:76) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.commands.exception.CommandException.of(CommandException.java:27) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.client.BridgeInternalServiceClient.lambda$invokeCommand$2(BridgeInternalServiceClient.java:93) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:239) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:229) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:130) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.set(AbstractFuture.java:664) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractCatchingFuture$CatchingFuture.setResult(AbstractCatchingFuture.java:210) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractCatchingFuture.run(AbstractCatchingFuture.java:138) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:112) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.SettableFuture.setException(SettableFuture.java:54) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.client.BridgeServiceTaskRetrier.onFailure(BridgeServiceTaskRetrier.java:102) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1228) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:466) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:442) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) ~[fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) ~[fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_131]
Caused by: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	... 13 common frames omitted
2017-11-14 17:33:55,586 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker0] Received incoming command response for unknown request ID 0. This is usually because a response was received after the command timed out locally.
 in  -[WorkerLogger:Unity]
2017-11-14 17:33:56,876 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker1] Received incoming command response for unknown request ID 2. This is usually because a response was received after the command timed out locally.
 in  -[WorkerLogger:Unity]
2017-11-14 17:34:14,350 WARN  [i.b.logging.EngineLogMessageHandler] [] [] [] : [Worker: UnityWorker0] NPC failed to receive Harvest response
 in  -[WorkerLogger:Unity]
2017-11-14 17:34:14,538 INFO  [i.b.client.BridgeServiceTaskRetrier] [] [] [] : An internal operation forwarding EntityCommandRequest for "entity_id: 1007
component_id: 1601
" failed on the SpatialOS side of the connection to UnityWorker0 after 0 retries, is the deployment under heavy load?
io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:442) [fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) [fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) [fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
2017-11-14 17:34:14,541 WARN  [i.b.commands.BridgeCommandHandler] [] [] [] : Failed to execute command entity_command_request {
  endpoint {
    entity_id: 1007
    component_id: 1601
  }
  payload: "\212d\004\n\002\b\017"
}
 on UnityWorker0
improbable.bridge.commands.exception.CommandException: DEADLINE_EXCEEDED
	at improbable.bridge.commands.exception.CommandException.ofStatusRuntimeException(CommandException.java:76) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.commands.exception.CommandException.of(CommandException.java:27) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.client.BridgeInternalServiceClient.lambda$invokeCommand$2(BridgeInternalServiceClient.java:93) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:239) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture$TransformFuture.doTransform(AbstractTransformFuture.java:229) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:130) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.set(AbstractFuture.java:664) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractCatchingFuture$CatchingFuture.setResult(AbstractCatchingFuture.java:210) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractCatchingFuture.run(AbstractCatchingFuture.java:138) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractTransformFuture.run(AbstractTransformFuture.java:112) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.SettableFuture.setException(SettableFuture.java:54) ~[fabric__runtime_deploy.jar:na]
	at improbable.bridge.client.BridgeServiceTaskRetrier.onFailure(BridgeServiceTaskRetrier.java:102) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1228) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:911) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.complete(AbstractFuture.java:822) ~[fabric__runtime_deploy.jar:na]
	at com.google.common.util.concurrent.AbstractFuture.setException(AbstractFuture.java:686) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$GrpcFuture.setException(ClientCalls.java:466) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:442) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) ~[fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) ~[fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_131]
Caused by: io.grpc.StatusRuntimeException: DEADLINE_EXCEEDED
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	... 13 common frames omitted
2017-11-14 17:34:26,136 INFO  [i.c.discovery.ServiceDiscoveryImpl] [] [] [] : Shutting down all services
2017-11-14 17:34:26,141 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.StatusDiscoveryInternalService) on 127.0.0.1:5301.
2017-11-14 17:34:26,145 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Stopped Service(improbable.platform.runtime.ModuleService) on 127.0.0.1:9000.
2017-11-14 17:34:26,146 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Stopped Service(improbable.platform.runtime.HealthCheckService) on 127.0.0.1:9001.
2017-11-14 17:34:26,146 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.PerfInternalService) on 127.0.0.1:9002.
2017-11-14 17:34:26,146 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.ContentionInternalService) on 127.0.0.1:9003.
2017-11-14 17:34:26,147 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Stopped Service(improbable.platform.runtime.FeatureService) on 127.0.0.1:9004.
2017-11-14 17:34:26,147 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.NodeConfigurationInternalService) on 127.0.0.1:9005.
2017-11-14 17:34:26,147 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.WorldExtentsInternalService) on 127.0.0.1:9006.
2017-11-14 17:34:26,149 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.EntityIdInternalService) on 127.0.0.1:9007.
2017-11-14 17:34:26,155 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.WorkerPrestartRegistryInternalService) on 127.0.0.1:9008.
2017-11-14 17:34:26,162 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.RouterLookupInternalService) on 127.0.0.1:9009.
2017-11-14 17:34:26,165 WARN  [i.c.c.f.FlagzFieldRegistryUpdater] [] [] [] : Fabric flagz update stream failed.
io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:395) [fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) [fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) [fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[na:1.8.0_131]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[na:1.8.0_131]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_131]
	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_131]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_131]
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288) ~[fabric__runtime_deploy.jar:na]
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:372) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:123) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) ~[fabric__runtime_deploy.jar:na]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858) ~[fabric__runtime_deploy.jar:na]
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138) ~[fabric__runtime_deploy.jar:na]
	... 1 common frames omitted
2017-11-14 17:34:26,168 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.RouterInternalService) on 127.0.0.1:9010.
2017-11-14 17:34:26,166 WARN  [i.c.config.flagz.WorkerFlagzUpdater] [] [] [] : Worker flagz update stream failed.
io.grpc.StatusRuntimeException: UNAVAILABLE
	at io.grpc.Status.asRuntimeException(Status.java:543) ~[fabric__runtime_deploy.jar:na]
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:395) [fabric__runtime_deploy.jar:na]
	at io.grpc.ForwardingClientCallListener.onClose(ForwardingClientCallListener.java:56) [fabric__runtime_deploy.jar:na]
	at me.dinowernli.grpc.prometheus.MonitoringClientCallListener.onClose(MonitoringClientCallListener.java:50) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:426) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl.access$100(ClientCallImpl.java:76) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.close(ClientCallImpl.java:512) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl.access$700(ClientCallImpl.java:429) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:544) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:52) [fabric__runtime_deploy.jar:na]
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:117) [fabric__runtime_deploy.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) [na:1.8.0_131]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) [na:1.8.0_131]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_131]
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method) ~[na:1.8.0_131]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43) ~[na:1.8.0_131]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[na:1.8.0_131]
	at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[na:1.8.0_131]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[na:1.8.0_131]
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:288) ~[fabric__runtime_deploy.jar:na]
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:1100) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:372) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:123) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496) ~[fabric__runtime_deploy.jar:na]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) ~[fabric__runtime_deploy.jar:na]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858) ~[fabric__runtime_deploy.jar:na]
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:138) ~[fabric__runtime_deploy.jar:na]
	... 1 common frames omitted
2017-11-14 17:34:26,170 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.RemoteMessageInternalService) on 127.0.0.1:9011.
2017-11-14 17:34:26,171 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.LocalBrokerInternalService) on 127.0.0.1:9012.
2017-11-14 17:34:26,173 INFO  [i.c.c.f.FlagzFieldRegistryUpdater] [] [] [] : Exponential backoff of 0 seconds after retry 1 of refreshing Fabric flagz.
2017-11-14 17:34:26,173 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.BridgeOracleInternalService) on 127.0.0.1:9013.
2017-11-14 17:34:26,173 INFO  [i.c.config.flagz.WorkerFlagzUpdater] [] [] [] : Exponential backoff of 0 seconds after retry 1 of refreshing worker flagz.
2017-11-14 17:34:26,175 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.GlobalBrokerInternalService) on 127.0.0.1:9014.
2017-11-14 17:34:26,178 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.BridgeServiceOracleInternalService) on 127.0.0.1:9015.
2017-11-14 17:34:26,179 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.LocalSnapshotInternalService) on 127.0.0.1:9016.
2017-11-14 17:34:26,180 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.ScalingStatsInternalService) on 127.0.0.1:9017.
2017-11-14 17:34:26,182 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.EntityQueryInternalService) on 127.0.0.1:9018.
2017-11-14 17:34:26,183 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.BridgeInternalService) on 127.0.0.1:9019.
2017-11-14 17:34:26,185 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Stopped Service(improbable.platform.inner.worker.WorkerRunner) on 127.0.0.1:9020.
2017-11-14 17:34:26,186 INFO  [i.core.discovery.server.LocalServer] [] [] [] : Stopped Service(improbable.fabric.internal.service.ReceptionistInternalService) on 127.0.0.1:9021.
2017-11-14 17:34:26,186 INFO  [i.core.discovery.server.PublicServer] [] [] [] : Stopped Service(improbable.platform.runtime.ReceptionistService) on 127.0.0.1:7777.
